integration-cli/docker_cli_pull_test.go

```diff
@@ -369,3 +369,40 @@ func (s *DockerTrustSuite) TestTrustedPullWithExpiredSnapshot(c *check.C) {
 		}
 	})
 }
+
+// #15589
+func (s *DockerTrustSuite) TestPullCancellation(c *check.C) {
```

Comment:
As someone who is interested in pull cancellation (e.g, ctrl-c stops the download), I am concerned that `TestPullCancellation` is a confusing name, since I assume that this pull request is not related to stopping the download (as build cancellation does analogously stop the build). If that's the case, can it be renamed?

Suggestions:
```
TestPullClientDisconnect
TestPullClientGiveup
TestPullClientGoaway
```

In addition, a comment above the function name explaining what is being tested would be a good idea. I'm just picturing a future developer making assumptions about what is being tested by this function and tying themselves in a knot.

I followed the issue number and that isn't quite enough to know what is being tested here.